src/test/java/org/folio/rest/api/CheckInStorageApiTest.java

```diff
@@ -171,12 +171,46 @@ public void cannotGetCheckInByIdIfDoesNotExists() throws InterruptedException,
     assertThat(checkInById.getStatusCode(), is(404));
   }
 
+  @Test
+  public void canCreateRecordWithoutOptionalParameters() throws InterruptedException,
+    ExecutionException, TimeoutException, MalformedURLException {
+
+    JsonObject checkInToCreate = createSampleCheckIn()
+      .withItemStatus(null)
+      .withItemLocationId(null)
+      .withRequestQueueSize(null)
+      .create();
+
+    IndividualResource createResult = checkInClient.create(checkInToCreate);
+
+    assertThat(createResult.getJson(), is(checkInToCreate));
+  }
+
+  @Test
+  public void cannotCreateRecordWithNegativeRequestQueueSize() throws InterruptedException,
+    ExecutionException, TimeoutException, MalformedURLException {
+
+    JsonObject checkInToCreate = createSampleCheckIn()
+      .withRequestQueueSize(-1)
+      .create();
+
+    JsonResponse createResult = checkInClient.attemptCreate(checkInToCreate);
+
+    assertThat(createResult,
+      isValidationResponseWhich(hasMessage("must be greater than or equal to 0")));
+    assertThat(createResult,
+      isValidationResponseWhich(hasParameter("requestQueueSize", "-1")));
+  }
+
```

Comment:
Does it make sense to create a test with non-existent location id?

Reply:
Sure, added.